modules/dlm/src/main/java/org/elasticsearch/dlm/DataLifecycleService.java

```diff
@@ -178,12 +204,13 @@ void run(ClusterState state) {
     }
 
     private void maybeExecuteRollover(ClusterState state, DataStream dataStream) {
-        if (dataStream.isIndexManagedByDLM(dataStream.getWriteIndex(), state.metadata()::index)) {
+        Index writeIndex = dataStream.getWriteIndex();
+        if (dataStream.isIndexManagedByDLM(writeIndex, state.metadata()::index)) {
             RolloverRequest rolloverRequest = getDefaultRolloverRequest(dataStream.getName());
             transportActionsDeduplicator.executeOnce(
                 rolloverRequest,
-                ActionListener.noop(),
-                (req, reqListener) -> rolloverDataStream(rolloverRequest, reqListener)
+                new ErrorRecordingActionListener(writeIndex.getName(), errorStore::recordError, errorStore::clearRecordedError),
+                (req, reqListener) -> rolloverDataStream(writeIndex.getName(), rolloverRequest, reqListener)
             );
         }
```

Comment:
I think we should consider this a "success" too and clear the error store, I think this is also a successful DLM run for this index and any relevant errors should be removed.

Reply:
I was planning on handing the managed -> unmanaged case once we had the APIs for it (to listen for changes). But I've added a method to pre-emptively clear the error store in the meantime (actually, maybe we just stick with this as opposed to introducing more listeners)